test/suite_test.go

```diff
@@ -612,35 +632,43 @@ func (e *e2e) kubectlRunOperatorNS(args ...string) string {
 }
 
 const (
-	curlBaseCMD = "curl -ksfL --retry 5 --retry-delay 3 --show-error "
-	curlCMD     = curlBaseCMD + "-H \"Authorization: Bearer `cat /var/run/secrets/kubernetes.io/serviceaccount/token`\" "
-	metricsURL  = "https://metrics.security-profiles-operator.svc.cluster.local/"
-	curlSpodCMD = curlCMD + metricsURL + "metrics-spod"
-	curlCtrlCMD = curlCMD + metricsURL + "metrics"
+	curlBaseCMD    = "curl -ksfL --retry 5 --retry-delay 3 --show-error "
+	headerAuth     = "-H \"Authorization: Bearer `cat /var/run/secrets/kubernetes.io/serviceaccount/token`\" "
+	curlCMD        = curlBaseCMD + headerAuth
+	curlHTTPVerCMD = curlCMD + "-I -w '%{http_version}\n' -o/dev/null "
+	metricsURL     = "https://metrics.security-profiles-operator.svc.cluster.local/"
+	webhooksURL    = "https://webhook-service.security-profiles-operator.svc.cluster.local/"
+	curlSpodCMD    = curlCMD + metricsURL + "metrics-spod"
+	curlCtrlCMD    = curlCMD + metricsURL + "metrics"
 )
 
-func (e *e2e) getSpodMetrics() string {
+func (e *e2e) runAndRetryPodCMD(podCMD string) string {
 	r := rand.New(rand.NewSource(time.Now().UnixNano())) // #nosec
 	letters := []rune("abcdefghijklmnopqrstuvwxyz")
 	b := make([]rune, 10)
 	for i := range b {
 		b[i] = letters[r.Intn(len(letters))]
 	}
+	maxTries := 0
 	// Sometimes the metrics command does not output anything in CI. We fix
 	// that by retrying the metrics retrieval several times.
 	var output string
 	if err := spoutil.Retry(func() error {
-		output = e.kubectlRunOperatorNS("pod-"+string(b), "--", "bash", "-c", curlSpodCMD)
+		output = e.kubectlRunOperatorNS("pod-"+string(b), "--", "bash", "-c", podCMD)
 		if len(strings.Split(output, "\n")) > 1 {
 			return nil
 		}
 		output = ""
-		return fmt.Errorf("no metrics output yet")
+		return fmt.Errorf("no output from pod command")
 	}, func(err error) bool {
 		e.logf("retry on error: %s", err)
-		return true
+		if maxTries < 5 {
```

Comment:
We're also using the retry implementation within curl. Is that the intention?

Reply:
Reduced the command retries to 3.